kafka/README.md

```diff
@@ -65,7 +67,38 @@ channels:
         topic: 'my-specific-topic-name'
         partitions: 20
         replicas: 3
-        bindingVersion: '0.3.0'
+        topicConfiguration:
+          cleanup.policy: ["delete", "compact"]
+          retention.ms: 604800000
+          retention.bytes: 1000000000
+          delete.retention.ms: 86400000
+          max.message.bytes: 1048588
+        bindingVersion: '0.4.0'
+```
+<a name="topicConfiguration"></a>
+## TopicConfiguration Object
+
+This objects contains information about the API relevant topic configuration in Kafka.
+
+Field Name |  Type   |                                                                             Description                                                                              | Applicability [default] | Constraints
+---|:-------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------:|:-----------------------:|---
+<a name="topicConfigurationCleanupPolicy"></a>`cleanup.policy` |  array  |                            The [`cleanup.policy`](https://kafka.apache.org/documentation/#topicconfigs_cleanup.policy) configuration option.                          |        OPTIONAL         | array may only contain `delete` and/or `compact`
+<a name="topicConfigurationRetentionMs"></a>`retention.ms` | integer |                               The [`retention.ms`](https://kafka.apache.org/documentation/#topicconfigs_retention.ms) configuration option.                                    |        OPTIONAL         | Must at least be -1
```

Comment:
I know what you mean by the "Must at least be -1" constraint, but I think it's possibly a little confusing.

Perhaps we could just defer to the official Kafka docs?

Or if we need to specify something here, we could say something like:
> If set, this should be equal to `-1` to indicate that no time limit is applied, otherwise it should be a value greater than 0.

Reply:
I like the idea of just referring to the kafka docs :) less duplication and always correct!